### DISCUSSION TITLE
Inquiry on Using STL Files for Tissue-Specific MSH Mesh Creation in SimNIBS

### DISCUSSION THREAD

**Discussion by mengwy2024** [simnibs/simnibs#464] [7.Creating head models]
Dear SimNIBS Community,

I am wondering if it’s possible to import STL files directly into SimNIBS to create MSH meshes for electric stimulation simulations, with an STL file designated for each tissue. If so, could you please advise on the steps required to do this?

I have followed the process outlined here (https://simnibs.github.io/simnibs/build/html/tutorial/advanced/custom_meshing.html) to create a mesh from a custom NIfTI file. However, the generated model contains sharp and irregular areas, which may affect the accuracy of the simulation. Therefore, I would like to use an optimized, smoothed STL model directly for the subsequent simulations.

Any guidance on how to achieve this would be greatly appreciated!

Thank you very much for your time and assistance.

Best regards,  
Meng

**Comment by TorgeW** [comment]
SimNIBS doesn't accept STL files for the creation of the head model. 
You could use the surfaces to crate masks to create a nifti file with the segmentation which can then be used for custom meshing. This will most likely create a more smooth result when the surfaces in the nifti volume are more smooth.

**Reply by mengwy2024** [reply]
Thank you very much for your response and guidance.
I understand that SimNIBS does not accept STL files directly for creating head models. I will try the method, as you suggested.
If I encounter any issues during this process, I may reach out for further assistance. Thank you again for your help!
